include/IECoreScene/MeshAlgo.h

```diff
@@ -76,6 +76,9 @@ void reverseWinding( MeshPrimitive *meshPrimitive );
 /// vertex spacing, provided the UVs are well layed out.
 PointsPrimitivePtr distributePoints( const MeshPrimitive *mesh, float density = 100.0, const Imath::V2f &offset = Imath::V2f( 0 ), const std::string &densityMask = "density", const std::string &uvSet = "uv", const std::string &position = "P" );
 
+/// Segment a mesh in to N meshes based on the N unique values contained in the data argument.
+std::vector<MeshPrimitivePtr> segment( const MeshPrimitive *mesh, const IECore::Data *data, const PrimitiveVariable &primitiveVariable );
```

Comment:
Can we document the arguments here a bit more? After chatting with @donboie I understand them, but lets jot that down for the masses.

Reply:
Sorry, I'm arriving at this party after it's already finished. Unless I'm misunderstanding something, even with the additional documentation this doesn't seem like a very useable interface. The simple use case is wanting to segment a primitive based on a primitive variable, but even for this simple case we're forcing the user to provide this `data` argument. Is it possible to make that argument optional instead, and call it `subset` to make it a bit more self-explanatory?

```
std::vector<MeshPrimitivePtr> segment( const MeshPrimitive *mesh, const PrimitiveVariable &primitiveVariable, const Data *subset=nullptr );
```

Reply:
Thinking about this more, I guess the other thing we'd need in Gaffer was a way of generating a string name for each segment. Maybe this is why you think the caller will need to generate the `data` argument anyway, at the same time as they generate the names?

Reply:
> Thinking about this more, I guess the other thing we'd need in Gaffer was a way of generating a string name for each segment. Maybe this is why you think the caller will need to generate the data argument anyway, at the same time as they generate the names?

Yup - this was my thinking.  I was going to add a PrimitiveVariableAlgo to generate the the required *data*. I can do this and make the argument optional?

Reply:
Yeah, that makes sense I think - that and calling `data` something more descriptive, like `segmentValues` or `subset` maybe. Could this PrimitiveVariableAlgo thing optionally fill a vector of names as well?